src/create_order.rs

```diff
@@ -1,9 +1,236 @@
+use crate::big_decimal::{BigDecimal, WBalance};
 use crate::*;
+use near_sdk::env::current_account_id;
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::{ext_contract, is_promise_success, log, Gas};
+use std::ops::Range;
+
+mod contract_interfaces;
+
+const NO_DEPOSIT: u128 = 0;
+const GAS_FOR_BORROW: Gas = Gas(180_000_000_000_000);
+
+pub const REF_FINANCE: &str = "ref-finance-101.testnet";
+
+#[ext_contract(ext_market)]
+trait MarketInterface {
+    fn borrow(&mut self, amount: WBalance) -> PromiseOrValue<U128>;
+}
+
+#[ext_contract(ext_token)]
+trait NEP141Token {
+    fn ft_transfer_call(
+        &mut self,
+        receiver_id: AccountId,
+        amount: WBalance,
+        memo: Option<String>,
+        msg: String,
+    );
+
+    fn ft_transfer(&mut self, receiver_id: AccountId, amount: WBalance, memo: Option<String>);
+}
+
+#[ext_contract(ext_self)]
+trait ContractCallbackInterface {
+    fn swap_callback(
+        &mut self,
+        user: AccountId,
+        amount: WBalance,
+        order: Order,
+    ) -> PromiseOrValue<Balance>;
+}
+
+/// Message parameters to receive via token function call.
+#[derive(Serialize, Deserialize)]
```

Comment:
you need to create a separate file via "ref_finance.rs" and move all structures and interfaces move there.